pkg/cli/agent.go

```diff
@@ -0,0 +1,461 @@
+	}
+
+	// Display agents
+	fmt.Printf("%-40s %-15s %-20s %s\n", "ID", "State", "Hostname", "Version")
+	fmt.Println(strings.Repeat("-", 90))
+	for _, agent := range agents {
+		hostname := agent.Hostname
+		if len(hostname) > 20 {
+			hostname = hostname[:17] + "..."
+		}
+		fmt.Printf("%-40s %-15s %-20s %s\n",
+			agent.ID,
+			agent.ConnectedState,
+			hostname,
+			agent.Version)
+	}
+
+	return nil
+}
+
+func (a *AgentListCmd) runInteractive(ctx context.Context, f *factory.Factory, org string) error {
+	loader := func(page int) tea.Cmd {
+		return func() tea.Msg {
+			opts := buildkite.AgentListOptions{
+				Name:     a.Name,
+				Hostname: a.Hostname,
+				Version:  a.Version,
+				ListOptions: buildkite.ListOptions{
+					Page:    page,
+					PerPage: a.PerPage,
+				},
+			}
+
+			agents, resp, err := f.RestAPIClient.Agents.List(ctx, org, &opts)
+			items := make([]agent.AgentListItem, len(agents))
+
+			if err != nil {
+				return err
+			}
+
+			for i, a := range agents {
+				a := a
+				items[i] = agent.AgentListItem{Agent: a}
+			}
+
+			return agent.NewAgentItemsMsg(items, resp.LastPage)
+		}
+	}
+
+	model := agent.NewAgentList(loader, 1, a.PerPage)
+
+	p := tea.NewProgram(model, tea.WithAltScreen())
+
+	if _, err := p.Run(); err != nil {
+		os.Exit(1)
+	}
+
+	return nil
+}
+
+func (a *AgentStopCmd) Run(ctx context.Context, f *factory.Factory) error {
+	if err := validateConfig(f.Config); err != nil {
+		return err
+	}
+
+	// Process agents - handling both single and multiple agents
+	var agentIDs []string
+
+	// Check if reading from stdin
+	if bk_io.HasDataAvailable(os.Stdin) {
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			line := strings.TrimSpace(scanner.Text())
+			if line != "" {
+				agentIDs = append(agentIDs, line)
+			}
+		}
+		if err := scanner.Err(); err != nil {
+			return fmt.Errorf("error reading from stdin: %w", err)
+		}
+	} else if len(a.Agents) > 0 {
+		// Use provided arguments
+		agentIDs = a.Agents
+	} else {
+		return fmt.Errorf("must supply agents to stop")
+	}
+
+	if len(agentIDs) == 0 {
+		return fmt.Errorf("no agents to stop")
+	}
+
+	// Use TUI for bulk operations (multiple agents) if we have a TTY
+	if len(agentIDs) > 1 && isatty.IsTerminal(os.Stdout.Fd()) {
+		return a.runInteractiveStop(ctx, f, agentIDs)
+	}
+
+	// Single agent: stop without TUI
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(agentIDs))
+	sem := make(chan struct{}, a.Limit) // Limit concurrent operations
+
+	for _, agentArg := range agentIDs {
+		wg.Add(1)
+		go func(agentArg string) {
+			defer wg.Done()
+			sem <- struct{}{}        // Acquire semaphore
+			defer func() { <-sem }() // Release semaphore
+
+			org, agentID := parseAgentArg(agentArg, f.Config)
+
+			// Stop the agent
+			var err error
+			spinErr := bk_io.SpinWhile(fmt.Sprintf("Stopping agent %s", agentArg), func() {
+				_, err = f.RestAPIClient.Agents.Stop(ctx, org, agentID, a.Force)
+			})
+			if spinErr != nil {
+				errChan <- spinErr
```

Comment:
Could it be another race condition when multiple goroutines write to `errChan` and reading from shared state? Might be worth looking into https://pkg.go.dev/golang.org/x/sync/errgroup which provides a friendly abstraction over goroutines handling.